packages/sdk/src/browser-wallets/phantom/index.ts

```diff
@@ -0,0 +1,153 @@
+import { Psbt } from "bitcoinjs-lib";
+
+import { getAddressFormat } from "../../addresses";
+import { BrowserWalletNetwork } from "../../config/types";
+import {
+  BrowserWalletExtractTxFromNonFinalizedPsbtError,
+  BrowserWalletNetworkMismatchError,
+  BrowserWalletNotInstalledError,
+  OrditSDKError,
+} from "../../errors";
+import { BrowserWalletSignResponse, WalletAddress } from "../types";
+import { PhantomSignPSBTOptions } from "./types";
+
+function isInstalled(): boolean {
+  if (typeof window === "undefined") {
+    throw new OrditSDKError("Cannot call this function outside a browser");
+  }
+  return typeof window.phantom !== "undefined";
+}
+
+function validateExtension(network: BrowserWalletNetwork = "mainnet"): void {
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Phantom Wallet not installed");
+  }
+
+  if (network !== "mainnet") {
+    throw new BrowserWalletNetworkMismatchError(
+      "Phantom Wallet only supports mainnet",
+    );
+  }
+}
+
+async function getAddresses(
+  network: BrowserWalletNetwork = "mainnet",
+): Promise<WalletAddress[]> {
+  validateExtension(network);
+  const bitcoinAccounts = await window.phantom.bitcoin.requestAccounts();
+  const walletAddresses: WalletAddress[] = [];
+  const purposeList: string[] = [];
+
+  bitcoinAccounts.forEach((account) => {
+    purposeList.push(account.purpose);
+    walletAddresses.push({
+      publicKey: account.publicKey,
+      address: account.address,
+      format: getAddressFormat(account.address, network),
+    });
+  });
+
+  if (!purposeList.includes("ordinals")) {
+    throw new OrditSDKError("No taproot address found");
+  }
```

Comment:
This is not working because a segwit address is returned by phantom wallet and marked as "ordinals" in purpose if taproot isn't enabled

Reply:
updated